### PR TITLE
Human vs human

### PR DESCRIPTION
**Pull request by elizabethengelman**
completed Human vs Human story. Was not sure how to test the main game loop in core, or get-move in game if the user inputs an invalid move.

### CODE REVIEW

File: spec/tic_tac_toe_clojure/game_spec.clj [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+		(with-out-str (it)))
+
+	(describe "valid-move?"
+			(it "returns true if valid"
+				(let [current-board { 1 "" 2 "X" 3 "X" 4 "" 5 "" 6 "" 7 "" 8 "" 9 ""}]
+					(should= true (valid-move? 1 current-board))))
+		  (it "returns false if the space is already taken"
+		  	(let [current-board { 1 "" 2 "X" 3 "X" 4 "" 5 "" 6 "" 7 "" 8 "" 9 ""}]
+		  		(should= false (valid-move? 2 current-board))))
+			(it "returns false if the move is out of the scope of the board"
```
Comment (nhu313) [inline spec/tic_tac_toe_clojure/game_spec.clj:16]: You should add one more test for the upper bound.

File: spec/tic_tac_toe_clojure/game_spec.clj [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+				(let [current-board { 1 "" 2 "X" 3 "X" 4 "" 5 "" 6 "" 7 "" 8 "" 9 ""}]
+					(should= false (valid-move? 0 current-board)))))
+				
+  (describe "get-move"
+  	(it "gets the move of the current player if its a valid move"
+    	(let [current-board { 1 "" 2 "X" 3 "X" 4 "" 5 "" 6 "" 7 "" 8 "" 9 ""}]
+    		(should= 1 
+      		(with-in-str "1"
+      		(get-move 0 current-board)))))
+  	(it "continues to prompt player if they input an invalid move"))
```
Comment (nhu313) [inline spec/tic_tac_toe_clojure/game_spec.clj:26]: Did you forget to write a test for this?

File: spec/tic_tac_toe_clojure/game_spec.clj [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  (describe "game-outcome"
+  	(it "returns that player 1 wins"
+  		(let [current-board { 1 "X" 2 "X" 3 "X" 4 "" 5 "" 6 "" 7 "" 8 "" 9 ""}]
+  			(should= "Player 1 wins! Way to go X's!" (game-outcome current-board))))
+  	(it "returns that player 2 wins"
+  		(let [current-board { 1 "O" 2 "O" 3 "O" 4 "" 5 "" 6 "" 7 "" 8 "" 9 ""}]
+  			(should= "Player 2 wins! Way to go O's!" (game-outcome current-board))))
+  	(it "returns that there's a tie"
+  		(let [current-board { 1 "O" 2 "X" 3 "O" 4 "X" 5 "O" 6 "X" 7 "O" 8 "O" 9 "X"}]
+  			(should= "Player 2 wins! Way to go O's!" (game-outcome current-board))))))
```
Comment (nhu313) [inline spec/tic_tac_toe_clojure/game_spec.clj:42]: Is this expected right now? I thought it's a tie?

File: spec/tic_tac_toe_clojure/rules_spec.clj [hunk 4]
```diff
@@ -0,0 +1,74 @@
+(ns tic-tac-toe_clojure.rules_spec
+  (:require [speclj.core :refer :all]
+            [tic_tac_toe_clojure.rules :refer :all]))
+
+(describe "tic_tac_toe_clojure.rules"
+
+	(describe "its-a-winner?"
+		(it "returns true if the current line is a winner"
+			(let [current-board { 1 "X" 2 "X" 3 "X" 4 "" 5 "" 6 "" 7 "" 8 "" 9 ""}]
+				(should= true (its-a-winner? current-board 0)))))
```
Comment (nhu313) [inline spec/tic_tac_toe_clojure/rules_spec.clj:10]: what does the 0 and 1 mean? which type of player?

Comment (elizabethengelman) [inline spec/tic_tac_toe_clojure/rules_spec.clj:10]: 0 is supposed to represent Player 1, and 1 is representing Player 2. I suppose this is a bit confusing, is there a better way to specify which player is which?

File: spec/tic_tac_toe_clojure/game_spec.clj (hunk 3 above)
Comment (elizabethengelman) [inline spec/tic_tac_toe_clojure/game_spec.clj:42]: Oops! Yep, that's supposed to be a tie.

File: spec/tic_tac_toe_clojure/game_spec.clj (hunk 2 above)
Comment (elizabethengelman) [inline spec/tic_tac_toe_clojure/game_spec.clj:26]: Yep! I moved this functionality to the get-player-move function in the Player file, and added a test.

File: spec/tic_tac_toe_clojure/game_spec.clj (hunk 1 above)
Comment (elizabethengelman) [inline spec/tic_tac_toe_clojure/game_spec.clj:16]: Will do, thanks! :)

**Pull request merged**